test/upgrade/upgrade.go

```diff
@@ -113,3 +116,26 @@ func createTestNamespace(t *testing.T, ns string) {
 		t.Fatalf("Couldn't create namespace %q: %v", ns, err)
 	}
 }
+
+func streamLogs(t *testing.T, clients *test.Clients, serviceName string) logstream.Canceler {
+	testStream := logstream.New(context.Background(), clients.KubeClient,
+		logstream.WithNamespaces(test.ServingFlags.TestNamespace),
+		logstream.WithLineFiltering(false),
+		logstream.WithPodPrefixes(serviceName))
+	testStreamCancel, err := testStream.StartStream(serviceName, t.Logf)
+	if err != nil {
+		t.Fatalf("Unable to stream logs from test namespace %s: %v", test.ServingFlags.TestNamespace, err)
+	}
+
+	sysStream := logstream.New(context.Background(), clients.KubeClient,
```

Comment:
The two streams can't share the same logstream base?

Reply:
What do you mean by logstream base? 
The first stream doesn't use line filtering and streams from test namespace.
The second stream uses line filtering and streams from system namespaces.
I don't see any easy way to re-use something.